presto-main/src/main/java/com/facebook/presto/execution/SetSessionTask.java

```diff
@@ -84,6 +85,12 @@ public String getName()
 
         String value = serializeSessionProperty(type, objectValue);
 
+        if ((propertyName.toString().equals(TASK_WRITER_COUNT)) && (Integer.bitCount(Integer.parseInt(value)) != 1)) {
```

Comment:
@martint Would you check my understanding? I guess the validation is here [`SystemSessionProperties.java#L131`](https://github.com/prestodb/presto/blob/f3a6c62a87a4f5345f6a34f32c0d1f0414fca9ef/presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java#L131). If so, this line is never called until users set session something before (first set session is not calling this line) and the `concurrency` is current value (not value users try to set).

There's similar validation in here [`TaskManagerConfig.java#L274`](https://github.com/prestodb/presto/blob/f9a287cca14bd3bda2a0e12d348c0771c3225767/presto-main/src/main/java/com/facebook/presto/execution/TaskManagerConfig.java#L274). In this case, this annotation is irrelevant because the value is set only while server start up.